### PR TITLE
Modify model classes to allow serialization and optimization with TorchScript

### PR DESCRIPTION
**Pull request by galbiati**
The model class definitions had minor issues that prevented compilation with `torch.jit.script`:

* `vgg16_bn` used `namedtuple` in `forward`
  * No features of `namedtuple` were actually being used, so now returns a regular tuple
* `BidirectionLSTM` used a `try`/`except` in `forward`
  * Instead of try/except there, added a method override in `get_recognizer` when GPU is enabled to inject the `rnn.flatten_parameters()` call
* `torch.jit.script` does not support `nn.AdaptiveAvgPool2d` with `None` dims
  * Replaced with `torch.mean(x, dim=3)`, which is equivalent at the relevant line

Other edits are just import ordering / dead import removes, line breaks, etc.

Using `torch.jit.script` and `torch.jit.optimize_for_inference` allowed me to run models about 14% faster on CPU in [Colab](https://colab.research.google.com/drive/1Nw5YdvaDWXuWlZjgjiys5PS80lsV4KCS?usp=sharing).

### CODE REVIEW

**Comment by rkcosmos** [comment]
thanks, I'll look into it.
